packages/babel-core/src/transformation/file/options/build-config-chain.js

```diff
@@ -13,12 +13,18 @@ const PACKAGE_FILENAME = "package.json";
 const BABELIGNORE_FILENAME = ".babelignore";
 
 function exists(filename) {
-  const cached = existsCache[filename];
-  if (cached == null) {
-    return existsCache[filename] = fs.existsSync(filename);
-  } else {
-    return cached;
+  if (existsCache[filename] == null) {
+    if (typeof fs.existsSync !== "function") {
```

Comment:
Not in the global context, as we don't know if people already have babelrc: false, but we could do it [here](https://github.com/danez/babel/blob/429ca92412dcadd566ef23e63f2e349e67a685de/packages/babel-core/src/transformation/file/options/build-config-chain.js#L36) as earliest point in this file.